nexus/src/external_api/http_entrypoints.rs

```diff
@@ -3226,11 +3443,44 @@ async fn instance_disk_list(
     apictx.external_latencies.instrument_dropshot_handler(&rqctx, handler).await
 }
 
+#[endpoint {
+    method = POST,
+    path = "/v1/instances/{instance}/disks/attach",
+    tags = ["instances"],
+}]
+async fn instance_disk_attach_v1(
+    rqctx: Arc<RequestContext<Arc<ServerContext>>>,
+    path_params: Path<params::InstancePath>,
+    query_params: Query<params::OptionalProjectSelector>,
+    disk_to_attach: TypedBody<params::DiskPath>,
```

Comment:
I want to acknowledge that using `DiskPath` here is weird. My plan is to eventually convert all the `*Path` structs into `*Identifier` for consistency. Whether they're used in the path or provided via the body doesn't really matter given it's just different places to specify a resource.